docs/manual/inference.tex

```diff
@@ -126,22 +92,209 @@
+  wpi.sh [-d PROJECTDIR] [-t TIMEOUT] -- [DLJC-ARGS]
+\end{Verbatim}
+
+Arguments in square brackets are optional.
+Here is an explanation of the arguments:
+
+\begin{description}
+\item[-d PROJECTDIR]
+  The top-level directory of the project.  It must contain an Ant, Gradle,
+  or Maven buildfile. The default is the current working directory.
+
+\item[-t TIMEOUT]
+  The timeout for running the checker, in seconds
+
+\item[DLJC-ARGS]
+  Arguments that are passed directly to
+  \ahref{https://github.com/kelloggm/do-like-javac}{do-like-javac}'s
+  \<dljc> program without
+  modification.  One argument is required: \<-\relax-checker>, which indicates
+  what type-checker to run (use the fully-qualified name of the checker).
+  The \ahrefforurl{https://github.com/kelloggm/do-like-javac}{documentation of do-like-javac}
+  describes the other commands that its WPI tool supports.
+\end{description}
+
+You may need to wait a few minutes for the command to complete.
+
+
+\sectionAndLabel{Requirements for whole-program inference scripts}{wpi-shared-requirements}
+
+The requirements to run \<wpi.sh> and \<wpi-many.sh> are the same:
+
+\begin{itemize}
+\item The project on which inference is run must contain an Ant, Gradle,
+  or Maven buildfile that compiles the project.
+\item \verb|JAVA8_HOME| environment variable must point to a Java 8 JDK.
+\item \verb|JAVA11_HOME| environment variable must point to a Java 11 JDK.
+\item \<CHECKERFRAMEWORK> environment variable must point to a built copy of the Checker Framework.
+\item Other dependencies:
+  ant,
+  awk,
+  curl,
+  git,
+  gradle,
+  mvn,
+  python2.7 (for dljc),
+  wget.
+
+  Python2.7 modules:
+  subprocess32.
+\end{itemize}
+
+
+
+\sectionAndLabel{Running whole-program inference on many projects}{wpi-many}
+
+The requirements to run \<wpi.sh> and \<wpi-many.sh> are the same. See Section~\ref{wpi-shared-requirements}
+for the list of requirements.
+
+To run an experiment on many projects:
+\begin{enumerate}
+\item Use \<query-github.sh> to search GitHub for candidate repositories.
+File \<securerandom.query> is an example query, and file
+\<securerandom.list>
+was created by running \<query-github.sh securerandom.query 100>. If you do
+not want to use GitHub, construct a file yourself that matches the format of
+the file \<securerandom.list>.
+
+\item Use \<wpi-many.sh> to run whole-program inference on every
+Ant, Gradle, or Maven project in a list of (GitHub repository URL, git hash)
+pairs.
+\begin{itemize}
+\item If you are using a checker that is distributed with the Checker
+Framework, use \<wpi-many.sh> directly.
+\item If you are using a checker that is not distributed with the Checker
+Framework (also known as a "custom checker"), file
+\<custom-checker-example.sh> is a no-arguments
+script that serves as an example of how to use \<wpi-many.sh>.
+\end{itemize}
+
+Log files are copied into a results directory.
+For a failed run, the log file indicates the reason that WPI could not
```

Comment:
This text belongs in the "single project" section, I think.  This text also differs slightly from what is already there.  Please move the text and reconcile the versions.

Reply:
It does not. `wpi-many.sh` processes the log files produced by `wpi-one.sh` and filters out those for which no actual results are available (and putting a log file with the reason a project was removed in the results directory). It also copies the `dlcj-out/wpi.log` file for projects for which results actually were available. This pre-processing makes it easier for users running large experiments to go through all the results quickly, by placing the top-level result for each project in a single directory.